packages/core/upload/server/controllers/admin-folder-file.js

```diff
@@ -60,25 +61,136 @@ module.exports = {
     await validateMoveManyFoldersFiles(body);
     const { folderIds = [], fileIds = [], destinationFolderId } = body;
 
-    const uploadService = getService('upload');
-    const folderService = getService('folder');
+    const trx = await strapi.db.transaction();
+    try {
+      // fetch folders
+      const existingFolders = await strapi.db
+        .queryBuilder(FOLDER_MODEL_UID)
+        .select(['id', 'uid', 'path'])
+        .where({ id: { $in: folderIds } })
+        .transacting(trx)
+        .forUpdate()
```

Comment:
It locks the database row so it cannot be read for other queries in parallele. Since we will update those folders, it is to avoid that a parallele query use this soon-outdated data (that can lead to break the consistency between `path` and `parent`).
I would have prefered to lock the database but knex doesn't allow it, so I use the transaction this way as it does an implicit lock.